src/main/java/com/backbase/ct/bbfuel/data/ProductSummaryDataGenerator.java

```diff
@@ -136,9 +138,12 @@ private static ArrangementsPostRequestBody getArrangementsPostRequestBody(String
             " " + currency + " " + bic.substring(0, 3) + accountNumber.substring(accountNumber.length() - 3);
         String fullArrangementName = currentAccountName + arrangementNameSuffix;
 
+        Set<String> legalEntityIds = new LinkedHashSet<String>(){};
+        legalEntityIds.add(externalLegalEntityId);
```

Comment:
Can be a simple `Collections.singleton(externalLegalEntityId)`

Reply:
changed